src/core/mesh/qgsmeshdataset.h

```diff
@@ -759,6 +760,49 @@ class CORE_EXPORT QgsMeshMemoryDatasetGroup: public QgsMeshDatasetGroup
     QVector<std::shared_ptr<QgsMeshMemoryDataset>> memoryDatasets;
 };
 
+/**
+ * \brief Class that represents a dataset with elevation value of the vertices of a existing mesh that can be edited
+ *
+ * \since QGIS 3.22
+ */
+class QgsMeshVerticesElevationDataset: public QgsMeshDataset
```

Comment:
These 2 classes are only to create DataBlocks? if yes, then it would be best to just create some utility functions, since you do not need a class for that? also in the utility functions you may look if the "Bed Elevation" groups does not exist and take that , and only if not try to contruct it (or alternatively we can fix MDAL so it always exists?)

Reply:
These two classes are like virtual dataset group, and must be derived class of QgsMeshDatasetGroup to be used in the renderer as dataset group